### PR TITLE
CB-46 limit comments only to logged users

### PR DESCRIPTION
**Pull request by Wojtach**
https://team-1626248039387.atlassian.net/jira/software/projects/CB/boards/1?selectedIssue=CB-46&text=react

login data: admin@gmail.com / admin

### CODE REVIEW

File: src/components/RecipeComments/RecipeComments.tsx [hunk 1]
```diff
@@ -27,7 +29,7 @@ export const RecipeComments: React.FC<RecipeCommentsProps> = ({
   const [showForm, setShowForm] = useState(false);
   const { comments, isLoading, onAdd, onDelete, onEdit } =
     useComments(commentsToDisplay);
-
+  const { user } = useAuth();
```
Comment (Aliath) [inline src/components/RecipeComments/RecipeComments.tsx:32]:
```suggestion
  const { user } = useAuth();
  const userAuthorized = !!user;
```

File: src/components/RecipeComments/RecipeComments.tsx [hunk 2]
```diff
@@ -39,15 +41,21 @@ export const RecipeComments: React.FC<RecipeCommentsProps> = ({
...
         </Grid>
         <Grid className="comments-grid__button-container" item sm={3}>
-          <Button
-            variant="contained"
-            color="primary"
-            startIcon={<AddIcon />}
-            disabled={showForm}
-            onClick={() => setShowForm(true)}
+          <Tooltip
+            title={user === null ? 'Only logged in users can add comments' : ''}
```
Comment (Aliath) [inline src/components/RecipeComments/RecipeComments.tsx:45]:
```suggestion
            title={userAuthorized ? '' : 'Only logged in users can add comments'}
```

File: src/components/RecipeComments/RecipeComments.tsx [hunk 3]
```diff
@@ -39,15 +41,21 @@ export const RecipeComments: React.FC<RecipeCommentsProps> = ({
...
           >
-            Add Comment
-          </Button>
+            <span>
+              <Button
+                variant="contained"
+                color="primary"
+                startIcon={<AddIcon />}
+                disabled={user === null || showForm}
```
Comment (Aliath) [inline src/components/RecipeComments/RecipeComments.tsx:52]:
```suggestion
                disabled={!userAuthorized || showForm}
```

**Review by Aliath — CHANGES_REQUESTED** [review]
Everything besides suggested changes is ok! 🚀

**Review by bartekguz — APPROVED** [review]

File: src/components/RecipeComments/RecipeComments.tsx [hunk 4]
```diff
@@ -30,6 +30,8 @@ export const RecipeComments: React.FC<RecipeCommentsProps> = ({
   const { comments, isLoading, onAdd, onDelete, onEdit } =
     useComments(commentsToDisplay);
   const { user } = useAuth();
+  const userIsAuthorized = !!user;
```
Comment (Aliath) [inline src/components/RecipeComments/RecipeComments.tsx:33]: `isUserAuthorized` 👀

**Review by Aliath — CHANGES_REQUESTED** [review]

**Review by Aliath — APPROVED** [review]

**Pull request merged**